modules/file-extension-common/src/main/java/org/mule/runtime/module/extension/file/api/metadata/FileAttributesMetadataResolver.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.file.api.metadata;
+
+import static org.mule.runtime.extension.api.introspection.declaration.type.ExtensionsTypeLoaderFactory.getDefault;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.metadata.MetadataContext;
+import org.mule.runtime.api.metadata.MetadataResolvingException;
+import org.mule.runtime.api.metadata.resolving.MetadataAttributesResolver;
+import org.mule.runtime.module.extension.file.api.FileSystem;
+
+import java.nio.file.FileSystemNotFoundException;
+import java.util.Optional;
+
+public class FileAttributesMetadataResolver implements MetadataAttributesResolver
+{
+
+    @Override
+    public MetadataType getAttributesMetadata(MetadataContext context, Object key) throws MetadataResolvingException, ConnectionException
+    {
+        Optional<FileSystem> connection = context.getConnection();
+        FileSystem fileSystem = connection.orElseThrow(() -> new FileSystemNotFoundException("Could not found file system to retrieve metadata from"));
+        return getDefault().createTypeLoader().load(fileSystem.getAttributesType());
```

Comment:
java the type loader as a field. Also don't use static import for the getDefault(). It becomes hard to understand without the context